### ISSUE TITLE
Style/AccessModifierDeclarations is catching named methods

### ISSUE THREAD

**Issue by jcoyne** [rubocop-hq/rubocop#6439]
When I have this code:
```ruby
      def foo; end
      private :foo
```

then rubycop reports this fault:
```
C: Style/AccessModifierDeclarations: private should not be inlined in method definitions.
  private :foo
  ^^^^^^^
``` 
--------

## Expected behavior
That is a named visibility declaration, not an inline method definition, so it should not raise a fault.

## Actual behavior

It calls a named visibility declaration an inline method definition and raises a fault.

## Steps to reproduce the problem

```ruby
      def foo; end
      private :foo
```

## RuboCop version

```
$ bundle exec rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.5.1 x86_64-darwin17)
```
